src/diffusers_interpret/plots/plots.py

```diff
@@ -0,0 +1,57 @@
+from typing import Literal, get_args, get_origin
+import inspect
+import pandas as pd
+import matplotlib.pyplot as plt
+%matplotlib inline
+
+def enforce_literals(function):
+    '''
+    Dynamically enforces the use of literals for function arguments.
+    Use the `inspect` module to get the previous frame to extract the provided arguments.
+    Use the function's attribute `__annotations__` to get the expected types.
+    '''
+    frame = inspect.stack()[1].frame
+    *_, parameters = inspect.getargvalues(frame)
+    for name, literal in function.__annotations__.items():
+        if get_origin(literal) is Literal and name in parameters:
+            value = parameters[name]
+            assert value in get_args(literal), f"'{value}' is invalid - valid options are {get_args(literal)}"
+
+# Plot types can only be one of these categories
+_TYPES = Literal['bar', 'barh', 'pie']
+
+def plot(self, type: _TYPES = 'bar', title: str = 'Token Attributions', rot: int = 60):
+    '''
+    Plot the normalized token attributes to have a comparative view.
+    Available plot types include bar chart, horizontal bar chart, and pie chart.
+    '''
+    # Dynamic assertation
+    enforce_literals(plot)
+
+    # Convert list of tuples to a dataframe
+    df = pd.DataFrame(self, columns=['Tokens', 'percent']).set_index('Tokens')
+
+    # Bar chart
+    if type == 'bar':
+        df.plot.bar(ylabel = 'percent',
+            title = title,
+            legend = False,
+            rot = rot);
+    
+    # Horizontal bar chart
+    elif type == 'barh':
+        df.plot.barh(ylabel = 'percent',
+            title = title,
+            legend = False,
+            rot = 0);
+
+    # Pie chart
+    elif type == 'pie':
+        df.plot.pie(y = 'percent',
+            title = title,
+            legend = False,
+            autopct = '%1.1f%%',
+            figsize = (8, 8));
```

Comment:
pandas is calling matplotlib in the background.
Let's call matplotlib directly and not add an extra dependency to the package

Note: haven't tested this code yet :)
```suggestion
    tokens, attributions = list(zip(*self)) # TODO: this can be changed, depending how we construct the class
    
    plot_kwargs = {'title': 'Token Attributions', **plot_kwargs} 
    if plot_type == 'bar':
        # Bar chart
        plt.bar(tokens, attributions, **{'legend': False, 'rot': 60, **plot_kwargs})
    
    elif plot_type == 'barh':
        # Horizontal bar chart
        plt.barh(tokens, attributions, **{'legend': False, 'rot': 0, **plot_kwargs})

    elif type == 'pie':
        # Pie chart
         plt.pie(attributions, **{'labels': tokens, **plot_kwargs})

    else:
        raise NotImplementedError(f"`plot_type={plot_type}` is not implemented. Choose one of: ['bar', 'barh', 'pie']")
    
```